crates/telemetry/src/lib.rs

```diff
@@ -100,10 +108,23 @@ pub fn init(spin_version: String) -> anyhow::Result<ShutdownGuard> {
 fn otel_error_handler(err: opentelemetry::global::Error) {
     static FIRST_OTEL_ERROR: std::sync::Once = std::sync::Once::new();
     FIRST_OTEL_ERROR.call_once(|| {
-        tracing::error!("There has been an error with the OpenTelemetry system, traces and metrics are likely failing to export");
-        tracing::error!("Further OpenTelemetry errors will be logged at DEBUG level")
+        tracing::error!(?err, "OpenTelemetry error");
+        tracing::error!("There has been an error with the OpenTelemetry system, traces, logs and metrics are likely failing to export. Further OpenTelemetry errors will be logged at DEBUG level.");
     });
-    tracing::debug!(?err, "OpenTelemetry error");
+
+    let signal = match err {
+        opentelemetry::global::Error::Metric(_) => "metrics",
+        opentelemetry::global::Error::Trace(_) => "traces",
+        opentelemetry::global::Error::Log(_) => "logs",
+        _ => "unknown",
+    };
+    metrics::monotonic_counter!(spin.otel_error_count = 1, signal = signal);
+
+    // Rate limit the logging of the OTel errors
+    if LAST_OTEL_ERROR.get().elapsed() > OTEL_ERROR_INTERVAL {
+        LAST_OTEL_ERROR.set(Instant::now());
+        tracing::debug!(?err, "OpenTelemetry error");
```

Comment:
Given the rate limiting we could probably bump this up to WARN? :thinking:

Reply:
Since we get a constant factor based on the number of threads it still spews quite a bit, but I like the idea of lowering it to WARN. I'll increase the `OTEL_ERROR_INTERVAL` so that it is less verbose and lower it to WARN.